### PR TITLE
Simplify packing coefficients

### PR DESCRIPTION
**Pull request by SarahRo**
- replace pack_coefficient_cell, pack_coefficient_exterior_facet and pack_coefficient_interior_facet by one function pack_coefficient_entity
- add test with non-constant coefficients
- simplify pack_coefficients

### CODE REVIEW

File: cpp/dolfinx/fem/utils.h [hunk 1]
```diff
@@ -455,72 +455,39 @@ void pack(const std::uint32_t cell, int bs, const xtl::span<T>& cell_coeff,
...
-
-    auto cell_coeff = c.subspan(index * cstride + offset, space_dim);
-    pack(cell, bs, cell_coeff, v, cell_info, dofmap, transformation);
-  }
-}
-
-// Pack a single coefficient for an exterior facet integral
-template <typename T>
-void pack_coefficient_exterior_facet(
+// Pack a single coefficient for a set of active entities
```
Comment (jorgensd) [inline cpp/dolfinx/fem/utils.h:458]:
```suggestion
/// Pack a single coefficient for a set of active entities
```

File: cpp/dolfinx/fem/utils.h [hunk 2]
```diff
@@ -455,72 +455,39 @@ void pack(const std::uint32_t cell, int bs, const xtl::span<T>& cell_coeff,
...
+/// @param[in, out] c The coefficient to be packed
+/// @param[in] cstride
+/// @param[in] v
```
Comment (jorgensd) [inline cpp/dolfinx/fem/utils.h:461]: Maybe add a sentence on each parameter about what they represent, i.e.
```
/// @param[in] cstride The number of coefficients to pack for each entity.
/// @param[in] v Describe whatever v is
/// @param[in] cell_info Array of bytes describing which permutation has to be applied on the cell to map it to the reference element.
```

Comment (SarahRo) [reply cpp/dolfinx/fem/utils.h:461]:
```suggestion
/// @param[in] cstride The total number of coefficient values to pack for each entity.
/// @param[in] v List of arrays with all degrees of freedom for the functions to be packed
/// @param[in] cell_info Array of bytes describing which transformation has to be applied on the cell to map it to the reference element.
```

Comment (SarahRo) [inline cpp/dolfinx/fem/utils.h:461]:
```suggestion
/// @param[in] cstride The total number of coefficient values to pack for each entity.
/// @param[in] v List of arrays with all degrees of freedom for the functions to be packed
/// @param[in] cell_info Array of bytes describing which transformation has to be applied on the cell to map it to the reference element.
```

Comment (SarahRo) [inline cpp/dolfinx/fem/utils.h:461]:
```suggestion
/// @param[in] cstride The total number of coefficient values to pack for each entity.
/// @param[in] v List of arrays with all degrees of freedom for the functions to be packed
```

File: cpp/dolfinx/fem/utils.h [hunk 3]
```diff
@@ -455,72 +455,39 @@ void pack(const std::uint32_t cell, int bs, const xtl::span<T>& cell_coeff,
...
+/// @param[in] cell_info
```
Comment (SarahRo) [inline cpp/dolfinx/fem/utils.h:462]:
```suggestion
/// @param[in] cell_info Array of bytes describing which transformation has to be applied on the cell to map it to the reference element.
```

**Review by jorgensd — APPROVED** [review]
Possibly merge in Garth's changes to avoid copying

**Review by garth-wells — APPROVED** [review]

**Pull request merged**